### PR TITLE
feat: add terradoc

### PR DESCRIPTION
**Pull request by kassianh**
Recreated README with terradoc, checked documentation for errors or inconsistencies.

### CODE REVIEW

**Review by katcipis — COMMENTED** [review]
A few comments but nothing major, overall seems OK but Im quite ignorant on this module :-)

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
       - id: terraform-validate
         exclude: ^examples|.terraform/
       - id: tflint
-      - id: gofmt
-      - id: goimports
       - id: golangci-lint
       - id: phony-targets
       - id: markdown-link-check
+        args: ['-p']   # When adding the -p flag, markdown-link-check will always with an exit code 0, even if dead links are found
+        verbose: true  # Forces the output of the hook to be printed even when the hook passes.
```
[the remark this answers is not in the document]
Comment (mariux) [reply .pre-commit-config.yaml:13]: why should indent be wrong? looks good.. the proposed indent won't work as a key is missing?

File: README.tfdoc.hcl [hunk 2]
```diff
@@ -0,0 +1,932 @@
...
+        Bucket Public Access Blocking,
+        S3 Access Points
+
+      - **Additional Features**:
+        Cross-Account access policy with forced `bucket-owner-full-control` ACL for direct access,
+        Create Cloudfront Origin Access Identity (OAI) and grant read-only access,
+        Grant read-only access to existing Cloudfront Origin Access Identity (OAI),
+        Allow ELB log delivery
+
+      - *Features not yet implemented*:
```
[the remark this answers is not in the document]
Comment (mariux) [reply README.tfdoc.hcl:82]: no it is supposed to be italic

File: README.tfdoc.hcl [hunk 3]
```diff
@@ -0,0 +1,932 @@
...
+    content = <<-END
+      See [variables.tf] and [examples/] for details and use-cases.
+    END
+
+    section {
+      title = "Top-level Arguments"
+
+      section {
+        title = "Module Configuration"
+
```
Comment (mariux) [inline README.tfdoc.hcl:124]: should be last in section

File: README.tfdoc.hcl [hunk 4]
```diff
@@ -0,0 +1,932 @@
...
+          type        = list(any)
+          description = <<-END
+            A list of dependencies. Any object can be _assigned_ to this list to define a hidden external dependency.
+          END
+        }
+
+        section {
+          title = "Bucket Configuration"
+
+          variable "bucket" {
```
Comment (mariux) [inline README.tfdoc.hcl:152]: this is main resource section .. please shift up one level..

File: README.tfdoc.hcl [hunk 5]
```diff
@@ -0,0 +1,932 @@
...
+                description = <<-END
+                  Specifies the Amazon S3 storage class to which you want the noncurrent versions object to transition.
+                  Can be `ONEZONE_IA`, `STANDARD_IA`, `INTELLIGENT_TIERING`, `GLACIER`, or `DEEP_ARCHIVE`.
+                END
+              }
+            }
+          }
+        }
+
+        section {
```
Comment (mariux) [inline README.tfdoc.hcl:477]: this and following subsections should go below extended resource configurations..

File: README.tfdoc.hcl [hunk 6]
```diff
@@ -0,0 +1,932 @@
...
+          Note: The AWS CloudFront allows specifying S3 region-specific endpoint when creating S3 origin,
+          it will prevent redirect issues from CloudFront to S3 Origin URL.
+        - **`hosted_zone_id`**: The Route 53 Hosted Zone ID for this bucket's region.
+        - **`region`**: The AWS region this bucket resides in.
+      - **`bucket_policy`**: All bucket policy object attributes as returned by the
+      `s3_bucket_policy` resource.
+      - **`origin_access_identity`**: All cloudfront origin access identity object attributes as returned by the
+      `aws_cloudfront_origin_access_identity` resource.
+      - **`access_point`**: A list of
+      `aws_s3_access_point` objects keyed by the `name` attribute.
```
Comment (mariux) [inline README.tfdoc.hcl:748]: please fix line breaks

File: README.tfdoc.hcl [hunk 7]
```diff
@@ -0,0 +1,932 @@
...
+    END
+  }
+
+  section {
+    title = "External Documentation"
+
+    section {
+      title   = "AWS Documentation S3"
+      content = <<-END
+        - Access Points: https://docs.aws.amazon.com/AmazonS3/latest/dev/   access-points.html
```
Comment (mariux) [inline README.tfdoc.hcl:758]: please fix link

File: README.tfdoc.hcl [hunk 8]
```diff
@@ -0,0 +1,932 @@
...
+    title   = "About Mineiros"
+    content = <<-END
+      Mineiros is a [DevOps as a Service][homepage] company based in Berlin, Germany.
+      We offer commercial support for all of our projects and encourage you to reach out if you have any questions or need help.
+      Feel free to send us an email at [hello@mineiros.io] or join our [Community Slack channel][slack].
+
+      We can also help you with:
+
+      - Terraform modules for all types of infrastructure such as VPCs, Docker clusters, databases, logging and monitoring, CI, etc.
+      - Consulting & training on AWS, Terraform and DevOps
```
Comment (mariux) [inline README.tfdoc.hcl:810]: please copy content for default sections from latest template..

**Review by mariux — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (soerenmartius) [reply .pre-commit-config.yaml:13]: indentation isn't wrong in this case :)  implicit keys need to be on a single line

The idea of having the CI pass is to show the developer that links are breaking but it shouldn't break the build (eg. sometimes that happens on purpose, eg. when a released hasn't been scheduled yet but the link was already implemented in the README.md to prepare the release).

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**